project-base/storefront/helpers/products/LastVisitedProducts.tsx

```diff
@@ -0,0 +1,34 @@
+import { usePersistStore } from 'store/zustand/usePersistStore';
+
+const LAST_VISITED_MAX_ITEMS = 20;
+
+export const GetLastVisitedProductCatalogNumbers = (): string[] => {
+    const lastVisitedProducts = usePersistStore((store) => store.lastVisitedProducts);
+    return lastVisitedProducts ? JSON.parse(lastVisitedProducts) : [];
+};
+
+export const SetLastVisitedProductCatalogNumbers = (newLastVisitedItemCatalogNumber: string): void => {
+    let olderItemCatalogNumbers = GetLastVisitedProductCatalogNumbers();
+
+    const updateUserState = usePersistStore((store) => store.updateUserState);
+    const alreadyAdded = olderItemCatalogNumbers.some(
+        (oldItemCatalogNumber) => oldItemCatalogNumber === newLastVisitedItemCatalogNumber,
+    );
+
+    if (!alreadyAdded) {
+        olderItemCatalogNumbers = [newLastVisitedItemCatalogNumber, ...olderItemCatalogNumbers];
```

Comment:
Pushing does not work in this scenario? Have you tried?

Reply:
True fixed, in very old version it was array of all information. Then changed to catnum only.